tests/jobs/ahm_analysis/test_ahm_analysis.py

```diff
@@ -0,0 +1,418 @@
+    ],
+)
+@pytest.mark.usefixtures("setup_tmpdir")
+def test_list_active_observations(test_data_root, input_dir, expected_result, obs_key):
+    """test function creates a dataframe reporting active observations during update"""
+    test_data_dir = os.path.join(test_data_root, input_dir)
+    df_update_log = ahmanalysis.make_update_log_df(test_data_dir, obs_key)
+    result = ahmanalysis.list_active_observations(df_update_log)
+    assert result == expected_result
+
+
+@pytest.mark.parametrize(
+    "input_obs, expected_misfit, update_log",
+    [
+        (["RWI_3_OBS"], 2.0, {"status": {0: "Active", 1: "Inactive", 2: "Active"}}),
+        (
+            ["RWI_3_OBS", "OP_3_WWCT1"],
+            7.0 / 4.0,
+            {"status": {0: "Active", 1: "Inactive", 2: "Active", 3: "Active"}},
+        ),
+        (["OP_3_WWCT1"], 1.0, {"status": {0: "Active"}}),
+    ],
+)
+def test_list_observations_misfit(input_obs, expected_misfit, update_log):
+    """test function creates a dataframe
+    reporting misfit data for each obs vector except the ones with All_obs-"""
+    misfit_df = pd.DataFrame(
+        {
+            "MISFIT:RWI_3_OBS": [6],
+            "MISFIT:OP_3_WWCT1": [1],
+            "MISFIT:OP_3_WWCT2": [0.15],
+            "MISFIT:OP_3_WWCT3": [0.25],
+        }
+    )
+    update_log_df = pd.DataFrame(update_log)
+    misfitval = ahmanalysis.list_observations_misfit(
+        input_obs, update_log_df, misfit_df
+    )
+    assert misfitval == expected_misfit
+
+
+@pytest.mark.parametrize(
+    "input_parameter, expected_result",
+    [
+        ("PORO", [np.array(8 * [float(nr)]) for nr in range(5, 10)]),
+        ("PERMX", [np.array(8 * [float(nr)]) for nr in range(5)]),
+    ],
+)
+@pytest.mark.usefixtures("setup_tmpdir")
+def test_import_field_param(input_parameter, expected_result, grid_prop):
+    """test function reads field parameter files and creates a dataframe from it"""
+
+    def flatten(regular_list):
+        return [item for sublist in regular_list for item in sublist]
+
+    grid = EclGridGenerator.createRectangular((2, 2, 2), (1, 1, 1))
+    grid.save_EGRID("MY_GRID.EGRID")
+
+    for iens in range(5):
+        grid_prop("PERMX", iens, grid.getGlobalSize(), f"{iens}_PERMX_field.grdecl")
+        grid_prop("PORO", iens + 5, grid.getGlobalSize(), f"{iens}_PORO_field.grdecl")
+
+    files = [f"{ens_nr}_{input_parameter}_field.grdecl" for ens_nr in range(5)]
+    result = ahmanalysis._import_field_param("MY_GRID", input_parameter, files)
+    assert flatten(result) == flatten(expected_result)
+
+
+def test_calc_delta_grid():
+    """test function creates a dataframe reporting mean
+    delta grids for field parameters"""
+    all_input_post = [
+        [1 + i for i in range(8)]
+        + [10 + i for i in range(8)]
+        + [20 + i for i in range(8)],
+        [3 + i for i in range(8)]
+        + [30 + i for i in range(8)]
+        + [40 + i for i in range(8)],
+    ]
+    all_input_prior = [
```

Comment:
Should we include input values that would benefit from the `np.absolute()`?

Reply:
I think we should also modify the numbers slightly. As it is now, the `groupby` for instance doesn't mean anything, it could just as well be "pick the first layer". The difference in these numbers will be a constant (e.g. 2) for the entire range. `(2 + i)*2` etc would suffice.

Reply:
done